message/messaging_service.cc

```diff
@@ -384,7 +421,18 @@ messaging_service::messaging_service(config cfg, scheduling_config scfg, std::sh
         _connection_index_for_tenant.push_back({_scheduling_config.statement_tenants[i].sched_group, i});
     }
 
-    register_handler(this, messaging_verb::CLIENT_ID, [] (rpc::client_info& ci, gms::inet_address broadcast_address, uint32_t src_cpu_id, rpc::optional<uint64_t> max_result_size) {
+    register_handler(this, messaging_verb::CLIENT_ID, [this] (rpc::client_info& ci, gms::inet_address broadcast_address, uint32_t src_cpu_id, rpc::optional<uint64_t> max_result_size, rpc::optional<utils::UUID> host_id) {
+        if (host_id) {
+            auto peer_host_id = locator::host_id(*host_id);
+            if (host_banned(peer_host_id)) {
+                throw rpc::drop_connection{};
```

Comment:
Can't you just close the connection here right away, why do you need to propagate the failure via an exception?